warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExpandMultiNormalizedTerms.java

```diff
@@ -329,29 +331,55 @@ protected JexlNode expandNodeForNormalizers(JexlNode node, Object data) {
                     Set<String> normalizedTerms = Sets.newHashSet();
                     List<JexlNode> normalizedNodes = Lists.newArrayList();
                     boolean failedNormalization = false;
+                    boolean regexNode = (node instanceof ASTNRNode || node instanceof ASTERNode);
                     // Build up a set of normalized terms using each normalizer
                     for (Type<?> normalizer : dataTypes) {
                         try {
-                            if (normalizer instanceof OneToManyNormalizerType && ((OneToManyNormalizerType<?>) normalizer).expandAtQueryTime()) {
-                                List<String> normTerms = ((OneToManyNormalizerType<?>) normalizer).normalizeToMany(term);
-                                if (normTerms.size() == 1) {
-                                    String normTerm = normTerms.iterator().next();
+                            if (regexNode) {
+                                String normTerm = normalizer.normalizeRegex(term);
+                                if (!normalizedTerms.contains(normTerm)) {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("normalizedTerm = " + normTerm);
+                                    }
                                     normalizedTerms.add(normTerm);
-                                    normalizedNodes.add(JexlNodeFactory.buildUntypedNode(node, fieldName, normTerm));
-                                } else {
-                                    List<JexlNode> normalizedOneToManyNodes = Lists.newArrayList();
-                                    Iterator<String> iter = normTerms.iterator();
-                                    while (iter.hasNext()) {
-                                        normalizedOneToManyNodes.add(JexlNodeFactory.buildUntypedNode(node, fieldName, iter.next()));
+                                    JexlNode normalizedNode = JexlNodeFactory.buildUntypedNode(node, fieldName, normTerm);
+                                    if ((!term.equals(normTerm)) && normalizer.normalizedRegexIsLossy(term)) {
+                                        JexlNode evalOnly = QueryPropertyMarker.create(JexlNodeFactory.buildUntypedNode(node, fieldName, term),
```

Comment:
we need to check if the term is indexOnly before adding EVAL_ONLY node

Reply:
done